### ISSUE TITLE
Issue with "aria-description": Missing a caveat about usage

### ISSUE THREAD

**Issue by norfali** [mdn/content#14478]
MDN URL: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-description

#### What information was incorrect, unhelpful, or incomplete?

`aria-description` isn't in 1.1 (as linked from the page) or 1.2 spec. Axe Linter are also flagging this up as invalid aria property.

Although this is available in WAI-ARIA 1.3, the spec is still a draft and not yet released.

Should this page not have a note/warning at the top that this isn't for production (yet)?

#### Specific section or headline?

Whole page.

#### What did you expect to see?

Do not use `aria-desciption` in production as the property isn't available yet. This also avoids engineers from adding this in their products thinking that it's fully supported like `aria-describedby`.

#### Did you test this? If so, how?

I was told that this property was supported in VO but I haven't been able to verify.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/accessibility/aria/attributes/aria-description`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-description
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/accessibility/aria/attributes/aria-description/index.md
* Last commit: https://github.com/mdn/content/commit/8ef4dbee26d302b627b6ce38e51531dc578de199
* Document last modified: 2022-03-18T16:38:43.000Z

</details>

**Comment by danielhjacobs** [comment]
FWIW, aria-description is supported in Firefox: https://hg.mozilla.org/integration/autoland/rev/fc4d5f72409c. However, it's not accessible from the DOM: https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaDescription. It is accessible from the DOM in Chrome/Edge/Opera/Samsung Internet. It's yet to be added to Safari: https://bugs.webkit.org/show_bug.cgi?id=206253.

It also seems axe core lets you configure what aria attributes to allow: https://github.com/dequelabs/axe-core/issues/2108.

**Comment by norfali** [comment]
@danielhjacobs, thanks for your response. So I guess this means that the article should follow the current specification and configuring axe-core doesn't feel like a feasible approach, right?

IMHO, a blurb or a paragraph at the top of the page warning that the `aria-description` property is not in the current WAI-ARIA spec might suffice.

**Comment by sideshowbarker** [comment]
> IMHO, a blurb or a paragraph at the top of the page warning that the `aria-description` property is not in the current WAI-ARIA spec might suffice.

To be clear about something: for MDN purposes in general, the only current specs are the versions hosted at w3c.github.io URLs — or otherwise at whatwg.org URLs or wicg.github.io URLs — but never the versions at w3.org URLs.

In some cases, the w3.org versions _may_ be in sync with with the current w3c.github.io sources — but not in all cases. Because of that, for MDN purposes, the w3.org versions are in general assumed to be possibly outdated and not current.

That means the maturity status of any particular W3C spec is irrelevant for MDN purposes.

So given all that, in the specific case of the ARIA spec, for MDN purposes it doesn’t matter at all whether or not something’s in the W3C ARIA 1.2 version — because that’s not the current spec; instead https://w3c.github.io/aria/ is the current spec.

What matters instead is what status of support for a particular feature is in implementations. And in cases (such as it sounds like this one maybe is) where there may not yet be reliable support for a particular feature across different implementations, then it’s useful to add a note to the relevant article saying as much.

But for MDN purposes, it’s not normally necessary or useful to add a note in any case where a particular feature hasn’t yet happened to be published yet in a spec version at a w3.org URL.

**Comment by norfali** [comment]
> > IMHO, a blurb or a paragraph at the top of the page warning that the `aria-description` property is not in the current WAI-ARIA spec might suffice.
> 
> To be clear about something: for MDN purposes in general, the only current specs are the versions hosted at w3c.github.io URLs — or otherwise at whatwg.org URLs or wicg.github.io URLs — but never the versions at w3.org URLs.
> 
> In some cases, the w3.org versions _may_ be in sync with with the current w3c.github.io sources — but not in all cases. Because of that, for MDN purposes, the w3.org versions are in general assumed to be possibly outdated and not current.
> 
> That means the maturity status of any particular W3C spec is irrelevant for MDN purposes.
> 
> So given all that, in the specific case of the ARIA spec, for MDN purposes it doesn’t matter at all whether or not something’s in the W3C ARIA 1.2 version — because that’s not the current spec; instead https://w3c.github.io/aria/ is the current spec.
> 
> What matters instead is what status of support for a particular feature is in implementations. And in cases (such as it sounds like this one maybe is) where there may not yet be reliable support for a particular feature across different implementations, then it’s useful to add a note to the relevant article saying as much.
> 
> But for MDN purposes, it’s not normally necessary or useful to add a note in any case where a particular feature hasn’t yet happened to be published yet in a spec version at a w3.org URL.

Thanks for the clarification, it does make sense.

Just for additional context, I think the confusion stemmed from:
* axe-core saying it is _not_ a valid ARIA property
* the reference link on the [page](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-description#specifications) points to a WAI-ARIA 1.1 spec which doesn't even mention `aria-description`
* In the same section, "Recommendation" was used for status.

So all of this detail indicates that teams can use `aria-description` like any other ARIA properties widely right now with no fear.

Based on your post, are we saying that the specification link in this article should be updated to point to https://w3c.github.io/aria/ and the onus is on the implementers to be aware of support? If that's case, I am ok with that.

Thanks again for the dialogue.

**Comment by sideshowbarker** [comment]
> are we saying that the specification link in this article should be updated to point to [w3c.github.io/aria](https://w3c.github.io/aria/)

Yes — and there’s actually already an open PR for making that change

> and the onus is on the implementers to be aware of support?

Yes

**Comment by norfali** [comment]
> Yes — and there’s actually already an open PR for making that change
> 
Ah I see the open PR - https://github.com/mdn/content/pull/13170 - looks good. Thank you for your help - appreciate what you do! 🙌 

Closing this.
